helixbass/tsc-rust
# Type-check bigint, heterogeneous union type

In this PR:
- parse + type-check bigint literals + type annotations
- type-check a "heterogeneous union" type

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
const a: bigint = 1n;

++a
```
then you shouldn't see any diagnostics debug-logged
If you run `cargo run tmp.tsx` against a source file containing eg:
```
const a: bigint = 1;

++a
```
then you should see a diagnostic debug-logged with message `Type 'number' is not assignable to type 'bigint'.`
If you run `cargo run tmp.tsx` against a source file containing eg:
```
const a: number = 1n;

++a
```
then you should see a diagnostic debug-logged with message `Type 'bigint' is not assignable to type 'number'.`
If you run `cargo run tmp.tsx` against a source file containing eg:
```
const a: number | bigint = 1;

++a
```
then you shouldn't see any diagnostics debug-logged
If you run `cargo run tmp.tsx` against a source file containing eg:
```
const a: boolean | number = 1;

++a
```
then you should see a diagnostic debug-logged with message `An arithmetic operand must be of type 'any', 'number', 'bigint' or an enum type.`

Based on `type-check-if-statement`
